source/extensions/tracers/xray/util.cc

```diff
@@ -0,0 +1,166 @@
+#include "extensions/tracers/xray/util.h"
+
+#include <algorithm>
+#include <chrono>
+#include <random>
+#include <regex>
+#include <vector>
+
+#include "common/common/hex.h"
+#include "common/common/utility.h"
+
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+void Util::mergeJsons(std::string& target, const std::string& source,
+                      const std::string& field_name) {
+  rapidjson::Document target_doc, source_doc;
+  target_doc.Parse(target.c_str());
+  source_doc.Parse(source.c_str());
+
+  target_doc.AddMember(rapidjson::StringRef(field_name.c_str()), source_doc,
+                       target_doc.GetAllocator());
+
+  rapidjson::StringBuffer sb;
+  rapidjson::Writer<rapidjson::StringBuffer> w(sb);
+  target_doc.Accept(w);
+  target = sb.GetString();
+}
+
+void Util::addArrayToJson(std::string& target, const std::vector<std::string>& json_array,
+                          const std::string& field_name) {
+  std::string stringified_json_array = "[";
```

Comment:
This function will be called when an array of child spans presented in json string format are added into an existing span presented in json string format . Since the child span already has string format, I don't want to change it into `kArrayType` object and then called `AddMember`. Instead, inside `addArrayToJson`, it calls `mergeJsons`, which parses into `rapidjson::Document` type and call `AddMember` once. It is just one way to implement.